presto-main/src/main/java/io/prestosql/sql/gen/BytecodeUtils.java

```diff
@@ -335,7 +335,7 @@ else if (type == Boolean.class) {
 
     public static BytecodeExpression invoke(Binding binding, String name)
     {
-        return invokeDynamic(BOOTSTRAP_METHOD, ImmutableList.of(binding.getBindingId()), name, binding.getType());
+        return invokeDynamic(BOOTSTRAP_METHOD, ImmutableList.of(binding.getBindingId()), name.replaceAll("[^(A-Za-z0-9_$)]", "_"), binding.getType());
```

Comment:
Should we escape the name instead to prevent conflicts?

Reply:
I would add a note that `name` here is for formatting purpose only.

How was it failing before this change?

Reply:
This is the error we got 
`java.lang.RuntimeException: Error compiling identity.function($operator$CAST(123)): invalid name: identity.function`

The actual reason was 
```
Caused by: java.lang.IllegalArgumentException: invalid name: identity.function
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:216)
	at io.airlift.bytecode.instruction.InvokeInstruction.checkUnqualifiedName(InvokeInstruction.java:441)
	at io.airlift.bytecode.instruction.InvokeInstruction.<init>(InvokeInstruction.java:319)
```

It looks like bytecode library doesn't support passing name with `.`

Reply:
Ah, so it seems there might be name conflict at byte level. I think it should be escaped then?

Maybe you add tests that confirms the conflict when `replaceAll` produces same name. I guess such functions need to be within same expression.

Reply:
IIRC since we use bindingId to resolve the `MethodHandle` for the scalar function the name shouldn't be a problem (even if it sees a similar scalar function name). I have updated the tests cases for the same and it wasn't showing any issues

Reply:
@electrum could you validate reasoning here?

@Praveen2112 I would still use escaping here as just to be safe as it's seem generated bytecode actually uses that method name

Reply:
At the bytecode level, invokedynamic takes a string as a name and any other arguments the caller wants to pass to the bootstrap function. The name can be anything, but our bytecode library wrapper in airlift is incorrectly, enforcing that it’d be a valid Java identifier. We should fix that constraint there.

Reply:
I was mistaken. There was some talk a few years ago about relaxing the constraints in the VM regarding how to formulate those identifiers to improve interop with other languages (which is the main reason invokedynamic was introduced in the first place), but the discussion went nowhere.

Reply:
so essentially this `name.replaceAll("[^(A-Za-z0-9_$)]", "_")` is the correct fix?

Reply:
Yes. If the name was significant, we’d mangle it in a way that was reversible, but we resolve the linkage based on an artificial “binding id”